ui/apps/platform/cypress/integration/collections/permissions.test.js

```diff
@@ -0,0 +1,116 @@
+import withAuth from '../../helpers/basicAuth';
+import { hasFeatureFlag } from '../../helpers/features';
+import { visit, visitWithStaticResponseForPermissions } from '../../helpers/visit';
+import navSelectors from '../../selectors/navigation';
+import { tryDeleteCollection, visitCollections } from './Collections.helpers';
+import { collectionSelectors } from './Collections.selectors';
+
+// Quick utility to create a basic collection for permission testing
+function createCollection(name) {
+    // Ignore autocomplete requests
+    // TODO Remove this once the feature is in
+    cy.intercept('/v1/collections/autocomplete', {});
+    localStorage.setItem('access_token', Cypress.env('ROX_AUTH_TOKEN'));
+    tryDeleteCollection(name);
+    visitCollections();
+    cy.get('a:contains("Create collection")').click();
+    cy.get('input[name="name"]').type(name);
+    cy.get('input[name="description"]').type('A collection for permission testing purposes');
+    cy.get('button:contains("All namespaces")').click();
+    cy.get('button:contains("Namespaces with names matching")').click();
+    cy.get('input[aria-label="Select value 1 of 1 for the namespace name"]').type('stackrox');
+    cy.get(`button:contains('stackrox')`).click();
+    cy.get('button:contains("Save")').click();
```

Comment:
Can be a follow up to write a helper function like `saveCollection` to encapsulate:
* click button
* wait on POST request

Reply:
In case you need or want to reuse a few collections for various tests, then if the helper function creates only if the collection does not already exist:
* It achieves reuse without coupling between tests or test files.
* It seems as if conditional create replaces need to try delete for view-only collections.

Reply:
Agreed, I'll switch this to an API based request.

I like the idea of deleting at the end of the test just for the sake of cleanup. I suppose in this case since the helper to create the function will always check existence first it isn't as much of a priority though.

Reply:
I moved this out to a helper function and verified that Cypress waits for the request to complete before proceeding, ince we can't `cy.wait()` on a request initiated via `cy.request()`.